cgmes/cgmes-conversion/src/main/java/com/powsybl/cgmes/conversion/Context.java

```diff
@@ -203,69 +205,132 @@ public PropertyBags phaseTapChangerTable(String tableId) {
+        @Override
+        public String toString() {
+            return description;
         }
+
+        private final String description;
+    }
+
+    public void invalid(String what, String reason) {
+        handleIssue(ConversionIssueCategory.INVALID, what, reason);
+    }
+
+    public void invalid(String what, Supplier<String> reason) {
+        handleIssue(ConversionIssueCategory.INVALID, what, reason);
     }
 
     public void ignored(String what, String reason) {
-        LOG.warn(IGNORED_REASON, what, reason);
+        handleIssue(ConversionIssueCategory.IGNORED, what, reason);
     }
 
     public void ignored(String what, Supplier<String> reason) {
-        if (LOG.isWarnEnabled()) {
-            LOG.warn(IGNORED_REASON, what, reason.get());
-        }
+        handleIssue(ConversionIssueCategory.IGNORED, what, reason);
     }
 
     public void pending(String what, Supplier<String> reason) {
-        if (LOG.isInfoEnabled()) {
-            LOG.info("PENDING {}. Reason: {}", what, reason.get());
-        }
+        handleIssue(ConversionIssueCategory.PENDING, what, reason);
     }
 
     public void fixed(String what, String reason) {
-        LOG.warn(FIXED_REASON, what, reason);
-    }
-
-    public void fixed(Supplier<String> what, String reason) {
-        if (LOG.isWarnEnabled()) {
-            LOG.warn(FIXED_REASON, what.get(), reason);
-        }
+        handleIssue(ConversionIssueCategory.FIXED, what, reason);
     }
 
     public void fixed(String what, Supplier<String> reason) {
-        if (LOG.isWarnEnabled()) {
-            LOG.warn(FIXED_REASON, what, reason.get());
-        }
+        handleIssue(ConversionIssueCategory.FIXED, what, reason);
     }
 
     public void fixed(String what, String reason, double wrong, double fixed) {
-        LOG.warn("Fixed {}. Reason: {}. Wrong {}, fixed {}", what, reason, wrong, fixed);
+        Supplier<String> reason1 = () -> String.format("%s. Wrong %.4f, was fixed to %.4f", reason, wrong, fixed);
+        handleIssue(ConversionIssueCategory.FIXED, what, reason1);
     }
 
     public void missing(String what) {
-        LOG.warn("Missing {}", what);
+        String reason1 = "";
+        handleIssue(ConversionIssueCategory.MISSING, what, reason1);
     }
 
     public void missing(String what, double defaultValue) {
-        LOG.warn("Missing {}. Used default value {}", what, defaultValue);
+        Supplier<String> reason1 = () -> String.format("Using default value %.4f", defaultValue);
+        handleIssue(ConversionIssueCategory.MISSING, what, reason1);
+    }
+
+    private void handleIssue(ConversionIssueCategory category, String what, String reason) {
+        if (reporter != null) {
+            reportIssue(category, what, reason);
+        }
+        logIssue(category, what, reason);
+    }
+
+    private void handleIssue(ConversionIssueCategory category, String what, Supplier<String> reason) {
+        if (LOG.isWarnEnabled() || reporter != null) {
```

Comment:
Report in any case. In default case, reporter should be `Reporter.NO_OP` meaning no operation is done

Reply:
I've kept the check for the reporter being different of `Reporter.NO_OP`. The check is done to avoid calling the `reason` supplier if the resulting message will not be used (an _optimization_ introduced some time ago).